### PR TITLE
Add UI improvements for Siddhi editor extension plate

### PR DESCRIPTION
**Pull request by SinthujanSintha**
## Purpose
Add the UI implementation for Siddhi editor extension plate improvement.

## Goals
Provide complete information in the Siddhi editor extension plate.

## Approach
In this PR I have, 
1. Improved HTML elements to show examples and syntax in the extension palette along with needed JS and CSS functionalities.
2. Remarkable JS is used to sanitise the data to convert MD files to HTML

## User stories
By adding new improvements in the Siddhi editor extension palette in the editor,user can directly get all the information which are related to each extension without referring Siddhi extension docs

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/css/styles.css [hunk 1]
```diff
@@ -1674,3 +1674,134 @@ div.dataTables_wrapper div.dataTables_info {
...
+/*model extra classes*/
+
+.extension > h4 {
+    font-weight: bold;
+    color: #b7b7b7;
+}
+
+div.extension-description > p > code,
+div.extension-description > p {
+    font-size: 17px;
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/css/styles.css]: Standardise font size, seems it common to all extension div

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/css/styles.css [hunk 2]
```diff
@@ -1674,3 +1674,134 @@ div.dataTables_wrapper div.dataTables_info {
...
+    color: #b7b7b7;
+}
+
+div.extension-description > p > code {
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/css/styles.css]: This also common to the entire div

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js [hunk 3]
```diff
@@ -126,7 +127,80 @@ define(['jquery', 'lodash', 'log', 'handlebar', 'designViewUtils', 'app/source-e
...
+        var pipelineNewlineEscape = function (data) {
+            return data.replace(/[|]/g, '&#124;').replace(/[\n]/g, '<br/>');
+        }
+
+        /**
+         * mdconvertion which converst md data into html.
+         * @param cloneOperator extension object
+         * @returns {*{}} return extension object which contains converted md
+         */
+        var mdConversion = function (cloneOperator) {
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]:
```suggestion
        var mdConversion = function (operator) {
```

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js [hunk 4]
```diff
@@ -126,7 +127,80 @@ define(['jquery', 'lodash', 'log', 'handlebar', 'designViewUtils', 'app/source-e
         };
 
         /**
-         * Flattens the metadata structure into an array ti reduce the search complexity.
+         * replacing the pipeline and newline character for md conversation.
+         * @param data string variable
+         * @returns {""} retrun string data
+         */
+        var pipelineNewlineEscape = function (data) {
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]:
```suggestion
        var normaliseString = function (data) {
```

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js [hunk 5]
```diff
@@ -126,7 +127,80 @@ define(['jquery', 'lodash', 'log', 'handlebar', 'designViewUtils', 'app/source-e
...
+            if (cloneOperator.description) {
+                cloneOperator.extensionDescription = markDownConvertor.render(cloneOperator.description);
+            }
+            if (cloneOperator.examples) {
+                cloneOperator.example = "";
+                cloneOperator.examples.forEach(function (e, i) {
+                    //change the "|" as "," and "\n" as "<br/> in returnAttributes description to avoid md
+                    // conversation bug.
+                    e.syntax = pipelineNewlineEscape(e.syntax);
+                    cloneOperator.example += "<h5>example " + (++i) + "</h5>"
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: Format this in multiple lines so it's easily readable, Same for all places you have string concats

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js [hunk 6]
```diff
@@ -126,7 +127,80 @@ define(['jquery', 'lodash', 'log', 'handlebar', 'designViewUtils', 'app/source-e
...
+                quotes: '“”‘’',
+                highlight: function (/*str, lang*/) {
+                    return '';
+                }
+            });
+            if (cloneOperator.description) {
+                cloneOperator.extensionDescription = markDownConvertor.render(cloneOperator.description);
+            }
+            if (cloneOperator.examples) {
+                cloneOperator.example = "";
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]:
```suggestion
                cloneOperator.combinedExamples = "";
```

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js [hunk 7]
```diff
@@ -126,7 +127,80 @@ define(['jquery', 'lodash', 'log', 'handlebar', 'designViewUtils', 'app/source-e
...
+                    //change the "|" as "," and "\n" as "<br/> in returnAttributes description to avoid md
+                    // conversation bug.
+                    e.syntax = pipelineNewlineEscape(e.syntax);
+                    cloneOperator.example += "<h5>example " + (++i) + "</h5>"
+                        + "<pre>" + e.syntax + "</pre>" + "<p>" + e.description + "</p>";
+                });
+                cloneOperator.example = markDownConvertor.render(cloneOperator.example);
+            }
+            if (cloneOperator.parameters) {
+                cloneOperator.parameterTable = "| Name | Possible DataTypes | Description | " +
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: Same

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js [hunk 8]
```diff
@@ -126,7 +127,80 @@ define(['jquery', 'lodash', 'log', 'handlebar', 'designViewUtils', 'app/source-e
...
+                    // conversation bug.
+                    m.description = pipelineNewlineEscape(m.description);
+                    cloneOperator.parameterTable += " | " + m.name + " | " + m.type.join('<br/>') + " | " +
+                        m.description + " | " + m.defaultValue + " | " + m.optional + " | "
+                        + m.dynamic + " | " + "\n";
+                });
+                cloneOperator.parameterTable = markDownConvertor.render(cloneOperator.parameterTable);
+            }
+            if (cloneOperator.returnAttributes) {
+                cloneOperator.returnTable = "| Name | DataTypes | Description |\n " +
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: Name is only available in some of the functions, do a check and do not add name column if this is not used. Refer to the old code where it compiled retyurn type tables

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js [hunk 9]
```diff
@@ -138,10 +212,10 @@ define(['jquery', 'lodash', 'log', 'handlebar', 'designViewUtils', 'app/source-e
...
                 if (meta.inBuilt.hasOwnProperty(type)) {
                     meta.inBuilt[type].forEach(function (operator) {
-                        var op = _.clone(operator);
-                        op.fqn = op.name;
-                        op.type = type;
-                        operators.push(op);
+                        var cloneOperator = _.clone(operator);
+                        cloneOperator.fqn = cloneOperator.name;
+                        cloneOperator.type = type;
+                        operators.push(mdConversion(cloneOperator));
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]:
```suggestion
                        operators.push(createMDFile(cloneOperator));
```

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js [hunk 10]
```diff
@@ -320,14 +394,15 @@ define(['jquery', 'lodash', 'log', 'handlebar', 'designViewUtils', 'app/source-e
...
          * @param index Operator index
-         * @param container Current container to find the context.
+         * @param exIndex clipboard syntax
+         * @param container container Current container to find the context
          */
-        OperatorFinder.prototype.copyToClipboard = function (index, container) {
+        OperatorFinder.prototype.copyToClipboard = function (index, exIndex, container) {
             if (this._operators[index]) {
-                var syntax = buildSyntax(this._operators[index]);
+                var syntax = this._operators[index].syntax[exIndex].clipboardSyntax;
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: Remove buildSyntax() function if is not used

**Comment by niveathika** [comment]
@SinthujanSintha Also it will be better to have some space after each examples

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/css/styles.css (hunk 1 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/css/styles.css]: 848bbc07f1a4d47336d5ee141b19c5ee5207128a this commit fix above need.

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/css/styles.css (hunk 2 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/commons/css/styles.css]: 848bbc0 this commit fix above need.

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js (hunk 3 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: 960249df47be3bbdc76d0016d7e1dd9ac324ac05 this commit fix above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js (hunk 4 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: 960249df47be3bbdc76d0016d7e1dd9ac324ac05 this commit fix above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js (hunk 5 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: 960249df47be3bbdc76d0016d7e1dd9ac324ac05 this commit fix above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js (hunk 6 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: 960249df47be3bbdc76d0016d7e1dd9ac324ac05 this commit fix above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js (hunk 7 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: 960249df47be3bbdc76d0016d7e1dd9ac324ac05 this commit fix above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js (hunk 8 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: 960249df47be3bbdc76d0016d7e1dd9ac324ac05 this commit fix above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js (hunk 9 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: 960249d this commit fix above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js (hunk 10 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/js/operator-finder/operator-finder.js]: 960249d this commit fix above request change

**Pull request merged**